### PR TITLE
[CDAP-18786] Respect locked fields when adding template

### PR DESCRIPTION
**Pull request by njbriggs**
# Respect locked fields when adding template

## Description
Respect locked fields when adding template to pipeline
Show locked icon for locked fields

## PR Type
- [X] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [ ] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [CDAP-18786](https://cdap.atlassian.net/browse/CDAP-18786)

## Test Plan
1. Tested with example in Jira
2. Tested with template with no locked fields
3. Tested with plugin directly; no template

## Screenshots

image

### CODE REVIEW

File: app/cdap/components/shared/ConfigurationGroup/PropertyRow/LockedTooltip.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import * as React from 'react';
+import withStyles, { WithStyles, StyleRules } from '@material-ui/core/styles/withStyles';
+import LockIcon from '@material-ui/icons/Lock';
+import PropertyRowTooltip from './PropertyRowTooltip';
+
+const styles = (theme): StyleRules => {
```
Comment (seanfeld) [inline app/cdap/components/shared/ConfigurationGroup/PropertyRow/LockedTooltip.tsx]: i know this is copied from above but should we replace this with styled components style css?

File: app/cdap/components/shared/ConfigurationGroup/PropertyRow/index.tsx [hunk 2]
```diff
@@ -208,25 +210,26 @@ class PropertyRowView extends React.Component<IPropertyRowProps, IState> {
...
             <MacroIndicator
               onClick={this.toggleMacro}
               disabled={disabled}
               isActive={this.state.isMacroTextbox}
             />
-          </If>
+          )}
+          {locked && <LockedTooltip />}
         </div>
-        <If condition={propertyLevelErrorMsg !== '' && widgetCategory !== PLUGIN}>
```
Comment (seanfeld) [inline app/cdap/components/shared/ConfigurationGroup/PropertyRow/index.tsx:222]: thanks for removing these

**Review by seanfeld — APPROVED** [review]
just one nit about styled components

File: app/cdap/components/shared/ConfigurationGroup/PropertyRow/LockedTooltip.tsx (hunk 1 above)
Comment (njbriggs) [reply app/cdap/components/shared/ConfigurationGroup/PropertyRow/LockedTooltip.tsx]: Converted for this and `DescriptionTooltip`. `PropertyRowTooltip` broke with styled components (some theming issue) and I don't have the bandwidth to troubleshoot it right now.

**Pull request merged**